toonz/sources/toonz/cellselection.cpp

```diff
@@ -2303,3 +2313,478 @@ void TCellSelection::renameMultiCells(QList<TXshCell> &cells) {
+    std::vector<TFrameId> fids;
+    m_sl->getFids(fids);
+    int i = m_rowIndex;
+    int f = 0;
+    while (i < m_frameCount + m_rowIndex) {
+      TFrameId fid = (fids.size() != 0) ? fids[f] : i;
+      TXshCell cell(m_sl.getPointer(), fid);
+      f++;
+      xsh->setCell(i, m_columnIndex, cell);
+      int appo = i++;
+      while (i < m_step + appo) xsh->setCell(i++, m_columnIndex, cell);
+    }
+    app->getCurrentScene()->notifySceneChanged();
+    app->getCurrentScene()->notifyCastChange();
+    app->getCurrentXsheet()->notifyXsheetChanged();
+  }
+
+  int getSize() const override { return sizeof *this; }
+  QString getHistoryString() override {
+    return QObject::tr("Create Level %1  at Column %2")
+        .arg(QString::fromStdWString(m_sl->getName()))
+        .arg(QString::number(m_columnIndex + 1));
+  }
+};
+
+//-----------------------------------------------------------------------------
+// Convert selected vector cells to ToonzRaster
+
+bool TCellSelection::areOnlyVectorCellsSelected() {
+  // set up basics
+  int r0, c0, r1, c1;
+  getSelectedCells(r0, c0, r1, c1);
+  int i;
+  ToonzScene *scene = TApp::instance()->getCurrentScene()->getScene();
+  TXsheet *xsh      = scene->getXsheet();
+
+  TXshCell firstCell = xsh->getCell(r0, c0);
+  if (firstCell.isEmpty()) {
+    DVGui::error(QObject::tr("This command only works on vector cells."));
+    return false;
+  }
+
+  TXshSimpleLevel *sourceSl = firstCell.getSimpleLevel();
+  if (sourceSl->getType() != PLI_XSHLEVEL) {
+    DVGui::error(QObject::tr("This command only works on vector cells."));
+    return false;
+  }
+
+  if (c0 != c1) {
+    DVGui::error(
+        QObject::tr("Please select only one column for this command."));
+    return false;
+  }
+
+  for (i = r0; i <= r1; i++) {
+    TXshCell testCell = xsh->getCell(i, c0);
+    if (testCell.isEmpty() || testCell.getSimpleLevel() != sourceSl) {
+      DVGui::error(
+          QObject::tr("All selected cells must belong to the same level."));
+      return false;
+    }
+  }
+  return true;
+}
+
+TXshSimpleLevel *TCellSelection::getNewToonzRasterLevel(
+    TXshSimpleLevel *sourceSl) {
+  ToonzScene *scene       = TApp::instance()->getCurrentScene()->getScene();
+  TFilePath sourcePath    = sourceSl->getPath();
+  std::wstring sourceName = sourcePath.getWideName();
+  TFilePath parentDir("+drawings");
+  TFilePath fp = scene->getDefaultLevelPath(TZP_XSHLEVEL, sourceName)
+                     .withParentDir(parentDir);
+
+  TFilePath actualFp = scene->decodeFilePath(fp);
+
+  int i                = 1;
+  std::wstring newName = sourceName;
+  while (TSystem::doesExistFileOrLevel(actualFp)) {
+    newName = sourceName + QString::number(i).toStdWString();
+    fp      = scene->getDefaultLevelPath(TZP_XSHLEVEL, newName)
+             .withParentDir(parentDir);
+    actualFp = scene->decodeFilePath(fp);
+    i++;
+  }
+  parentDir = scene->decodeFilePath(parentDir);
+  if (!TFileStatus(parentDir).doesExist()) {
```

Comment:
Here you are checking the existence of the parent directory.
I think this operation is not needed since the `parentDir` is always `+drawings` for now.
Even if you will change `parentDir` to any other non-existent path, it will surely be created on saving the level.